pkg/utils/addons/addons_test.go

```diff
@@ -0,0 +1,160 @@
+package addons
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	mf "github.com/manifestival/manifestival"
+	"gotest.tools/golden"
+)
+
+func TestCreatePipeline(t *testing.T) {
```

Comment:
Could we use test data table for those ? (aka a loop with `t.Run`)

Reply:
Thanks for pointing this, earlier we faced an issue while testing as the order was not maintained. But the new code takes care of this.